theano/tensor/signal/pool.py

```diff
@@ -255,6 +255,64 @@ def pool_3d(input, ws=None, ignore_border=None, stride=None, pad=(0, 0, 0),
     return output
 
 
+def spp_pooling(image, out_dimension, image_shape):
```

Comment:
Your code should work if image_shape isn't set. So here, do image_shape=None, and make that case work as we discussed.